tests/test_requests.py

```diff
@@ -626,6 +626,34 @@ def handler(request):
     assert request.raw_args == params
 
 
+def test_request_not_grouped_args(app):
+    # test multiple params with the same key
+    params = [('test', 'value1'), ('test', 'value2')]
+
+    @app.get("/")
+    def handler(request):
+        return text("pass")
+
+    request, response = app.test_client.get("/", params=params)
```

Comment:
Mind adding a test case to simulate the actual call with `?test=value1&test=value2` directly in the URL to your test case?

Reply:
:+1: